### PR TITLE
#160609544 integrate travis ci

### PR DESCRIPTION
**Pull request by MuhanguziDavid**
### What does this PR do?

Adds a travis.yml file to integrate the Github repo with Travis CI

### Description of tasks to be completed?

Currently, code changes cannot be automatically built and tested, and immediate feedback on the success of changes cannot be provided

This task adds a travis.yml file which enables integration with Travis for continuous integration to be enabled.

### How should this be manually tested?

Push changes to GitHub and the changes will be automatically built and tested. Feedback on success on changes will be shown in a badge in the ReadME.

### What are the relevant pivotal tracker stories?

[#160609544](https://www.pivotaltracker.com/story/show/160609544)

### Any background context you want to add?

N/A

### Screenshots
screen shot 2018-09-27 at 15 39 05
screen shot 2018-09-27 at 15 41 09

### CODE REVIEW

**Review by kakaemma — APPROVED** [review]
Nice work with the tests and continuous integration.

**Review by mendozabree — APPROVED** [review]
LGTM
Good work @MuhanguziDavid

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+install:
+  - pip install -r requirements.txt
+
+script:
+  - coverage run --source=authors/ manage.py test
+
+after_success:
+  - coveralls
+
+# deploy:
```
Comment (isheebo) [inline .travis.yml]: remove commented code if it is not needed

File: authors/apps/authentication/tests.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# tests for authentication application
+from django.test import TestCase
+from authors.apps.authentication.models import User, UserManager
+from rest_framework.test import APIClient
```
Comment (isheebo) [inline authors/apps/authentication/tests.py:3]: kindly sort these imports. starting with third-party library imports and then imports from your code.

Right clicking in the file in vscode and choosing `Sort Imports` can come in handy

File: authors/apps/authentication/tests.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+from rest_framework import status
+
+class ModelTestCase(TestCase):
+    """Class with tests to do with registration model"""
+    def setUp(self):
+        """Define the test client and test variables"""
+        self.old_count = User.objects.count()
+        self.user = User.objects.create_user(username="david", email="david@gmail.com", password="password")
+        # self.user_manager = UserManager(username="david", email="david@gmail.com", password="password")
```
Comment (isheebo) [inline authors/apps/authentication/tests.py]: remove un-unsed code

**Comment by isheebo** [comment]
@MuhanguziDavid please edit your commit message to follow the convention I gave yesterday to the dot.

I will leave a link to it here 👉 `https://github.com/andela/engineering-playbook/tree/e654a2dd79972dfe51a1b54d6c7ebc04d2c5e012/5.%20Developing/Conventions#commit-message`

**Comment by isheebo** [comment]
In the body of the `PR text`, when specifying, the pivotal tracker link, it would be much more useful if it is specified as a `Markdown` link, rather than `plain text`

File: authors/apps/authentication/tests.py (hunk 2 above)
Comment (MuhanguziDavid) [reply authors/apps/authentication/tests.py:3]: Thanks, I shall work on that

Comment (MuhanguziDavid) [reply authors/apps/authentication/tests.py:3]: The imports have been sorted

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+script:
+  - coverage run --source=authors/ manage.py test
+
+after_success:
+  - coveralls
+
+deploy:
+  provider: heroku
+  api_key:
+    secure: ebc475a8-e613-4b9c-8adc-58b408c546c3
```
Comment (isheebo) [inline .travis.yml]: you are exposing this to everyone. Please add this to the environmental variables on travis such that it can be read from there

**Comment by isheebo** [comment]
@MuhanguziDavid kindly resolve the merge conflicts here

**Comment by lym** [comment]
👍

**Pull request merged**